src/components/elements/inputs/masked-input.js

```diff
@@ -1,35 +1,53 @@
+import { Component } from 'react'
 import cx from 'classnames'
 import MaskedInput from 'react-text-mask'
 import createNumberMask from 'text-mask-addons/dist/createNumberMask.js'
 import emailMask from 'text-mask-addons/dist/emailMask.js'
 
-export default ({ className, errors, onCommit, onUpdate, property, value }) => {
-  const mask = getMask(property.format)
+export default class MaskedInputInput extends Component {
+  constructor(props) {
+    super()
+    this.state = { displayValue: props.value || ''}
+  }
+  render() {
+    const mask = this.getMask(this.props.property.format)
 
-  return (
-    <MaskedInput
-      className={cx(className, 'ct-input', 'ct-text-input')}
-      data-tip={errors}
-      readOnly={property.disabled}
-      id={property.name}
-      mask={mask}
-      onBlur={() => onCommit()}
-      onChange={e => onUpdate(e.target.value)}
-      title={property.title}
-      type={{ telephone: 'tel' }[property.display] || 'text'}
-      value={value} />
-  )
-}
+    return (
+      <MaskedInput
+        className={cx(this.props.className, 'ct-input', 'ct-text-input')}
+        data-tip={this.props.errors}
+        readOnly={this.props.property.disabled}
+        id={this.props.property.name}
+        mask={mask}
+        onBlur={() => this.props.onCommit()}
+        onChange={e => this.adjustAndUpdate(e.target.value)}
+        title={this.props.property.title}
+        type={{ telephone: 'tel' }[this.props.property.display] || 'text'}
+        value={this.state.displayValue} />
+    )
+  }
+
+  adjustAndUpdate(value) {
+    this.state.displayValue = value
+    this.setState(state)
```

Comment:
Yes you are both correct of course. Sorry! 
I am putting up a new PR really, really soon. I'll correct it there since I was so quick merging this.